### PR TITLE
docs: move to mkdocs

### PR DESCRIPTION
**Pull request by cpcloud**
This PR moves our documentation to [`mkdocs`](https://www.mkdocs.org/)
*including API documentation* by way of [`mkdocstrings`](https://mkdocstrings.github.io/)

The result is:

- API documentation and website are unified in single place with a single theme and experience
- A generally better looking website IMHO
- Faster UX, pretty much all around
- Search is fast and it looks nice

I also went through most public APIs in `ibis/expr/api.py` and added type
hints which show up very nice in the new docs.

You can run the docs from this PR using:

```
mkdocs serve
```

and clicking the link that shows up in the terminal.

### CODE REVIEW

File: docs/api/expressions/geospatial.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Geospatial Expressions
+
+<!-- ::: ibis.expr.types.GeoSpatialValue -->
+<!--  -->
+<!-- ## Column methods -->
+<!--  -->
+<!-- ::: ibis.expr.types.GeoSpatialColumn -->
```
Comment (cpcloud) [inline docs/api/expressions/geospatial.md:7]: TODO: I need to fix the docstrings for these before landing.

**Review by gforsyth — COMMENTED** [review]
I haven't looked over the deployment changes too closely, but a :heavy_plus_sign::100: on the look and feel when serving locally.  Transitions are nice and fast and everything feels much more connected.

**Review by gforsyth — APPROVED** [review]
Yeah, this looks great.  Everything is easier to read AND the rendered docs look better.  win-win

**Comment by cpcloud** [comment]
@gforsyth Thanks! That was definitely an explicit goal with this port.

**Comment by cpcloud** [comment]
@jreback Any comments here? There are no functional changes in this PR.

**Review by jreback — APPROVED** [review]
looks ok

**Comment by cpcloud** [comment]
Alright, squashing then merging! Thanks for the review

**Pull request merged**